src/Rector/Class_/PreferPHPUnitSelfCallRector.php

```diff
@@ -95,6 +98,21 @@ public function refactor(Node $node): ?Node
                 return null;
             }
 
+            $callerType = $this->nodeTypeResolver->getType($node->var);
+            if (! $callerType instanceof TypeWithClassName) {
+                return null;
+            }
+
+            $scope = $node->getAttribute(AttributeKey::SCOPE);
+            $methodReflection = $callerType->getClassReflection()?->getMethod($methodName, $scope);
```

Comment:
better to check `hasNativeMethod()` then `getNativeMethod()` later to avoid crash on not found:

```php
$classReflection = $this->reflectionResolver->resolveClassReflectionSourceObject($node);
if ($classReflection instanceof \PHPStan\Reflection\ClassReflection && $classReflection->hasNativeMethod($methodName)) {
    $method = $classReflection->getNativeMethod($methodName);
    if (! $method->isStatic()) {
        return null;
    }
}
```

Reply:
Thank you for your suggestion. I will push an updated pull request.